lib/testing/create-apollo-test-client.ts

```diff
@@ -0,0 +1,33 @@
+import { TestingModule } from '@nestjs/testing';
+import * as apolloServerTesting from 'apollo-server-testing';
+import { GraphQLModule, GraphQLFederationModule } from '..';
+import { ApolloServerBase } from 'apollo-server-core';
+
+/**
+ * Provides a single hook to run operations through the request pipeline,
+ * enabling the most thorough tests possible without starting up an HTTP server.
+ * @param testingModule the Nest.js testing module
+ * @returns a query and mutate function that can be used to run operations against the
+ * server instance. Currently, queries and mutations are the only operation
+ * types supported by createTestClient.
+ * @see https://www.apollographql.com/docs/apollo-server/testing/testing/
+ */
+export function createTestClient(
+  testingModule: TestingModule,
+): apolloServerTesting.ApolloServerTestClient {
+  return apolloServerTesting.createTestClient(getApolloServer(testingModule));
+}
```

Comment:
I would suggest removing this function as it adds a dependency on the `apollo-server-testing` library. Exposing the `getApolloServer()` (+ adding it to the docs and using in tests) should be sufficient.

Reply:
I moved it to `tests/utils` so it would only save repeated testing code.